crypto/ec/asm/ecp_nistz256-x86_64.pl

```diff
@@ -3618,17 +3618,26 @@ ()
 
 	or	$acc5, $acc4			# see if result is zero
 	or	$acc0, $acc4
-	or	$acc1, $acc4
+	or	$acc1, $acc4			# !is_equal(U1, U2)
 
-	.byte	0x3e				# predict taken
-	jnz	.Ladd_proceed$x			# is_equal(U1,U2)?
 	movq	%xmm2, $acc0
 	movq	%xmm3, $acc1
-	test	$acc0, $acc0
-	jnz	.Ladd_proceed$x			# (in1infty || in2infty)?
+
+	or	$acc0, $acc4
+	.byte	0x3e				# predict taken
+	jnz	.Ladd_proceed$x			# !is_equal(U1, U2) || in1infty || in2infty
+
+	# We now know A = B or A = -B and neither is infinity. Compare the
+	# y-coordinates via S1 and S2.
 	test	$acc1, $acc1
-	jz	.Ladd_double$x			# is_equal(S1,S2)?
+	jz	.Ladd_double$x			# is_equal(S1, S2)
```

Comment:
Not sure why the question mark is removed.  It symbolises that a question is asked, and indeed, `jz` asks if the zero flag is set (i.e. if the test shows equality)...

This goes for *all* such edits

Reply:
I think I did it because the comment on computing $acc4 and the sequence of ||s was awkward with the question marks, but honestly this was so long ago that I forget. I don't think the question marks add any value though.